mozetl/search/daily_search_rollups.py

```diff
@@ -1,309 +1,371 @@
 """
-# Daily main_summary to Vertica Search Rollups
+main_summary to Vertica Search Rollups
 
 This job is added by Bug 1364530 and was originally located
 [1]. Search rollups are computed and then ingested by Vertica.
 
-Please note that this job has not been reviewed or tested. Previous
-discussion here [2]. We have an open bugs for tests here [3].
-
 [1] https://gist.github.com/SamPenrose/856aa21191ef9f0de18c94220cd311a8
-[2] mozilla/mozilla-reports#23
-[3] https://bugzilla.mozilla.org/show_bug.cgi?id=1365653
 """
-import csv
-import datetime as DT
-import smtplib
-import subprocess
-import sys
-import time
-from email.message import Message
-
-from pyspark.sql import SparkSession
+
+import logging
+import re
+
+import arrow
+import boto3
 import click
+from pyspark.sql import SparkSession, functions as F
 
+from mozetl import utils
 
-READ_VERSION = 4
-READ_STEM = 's3://telemetry-parquet/main_summary/v{}/'.format(READ_VERSION)
-READ_TAIL = 'submission_date_s3={}/'
-DAY_READ_PATH = READ_STEM + READ_TAIL
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
 
-WRITE_VERSION = 1
-BASH_WRITE_STEM = 's3://net-mozaws-prod-us-west-2-pipeline-analysis/spenrose/search/to_vertica/'
-ROLLUP_WRITE_STEM = BASH_WRITE_STEM + 'daily/'
-MANIFEST_WRITE_STEM = BASH_WRITE_STEM + 'manifests/'
 
-PROFILE_DAY = 'profile_day'
-PROFILE_COLUMN = 'concat(client_id, submission_date) as {}'.format(PROFILE_DAY)
+def explode_searches(dataframe):
+    """Expand the search_count list for each ping"""
+    columns = [
+        "profile",
+        "country",
+        "locale",
+        "distribution_id",
+        "default_provider",
+    ]
+    exploded_search_columns = [
+        F.col("search_counts.engine").alias("search_provider"),
+        F.col("search_counts.count").alias("search_count")
+    ]
+
+    null_search_columns = [
+        F.lit("NO_SEARCHES").alias("search_provider"),
+        F.lit(0).alias("search_count"),
+    ]
+
+    # explode search counts
+    exploded_searches = (
+        dataframe
+        .withColumn("search_counts", F.explode("search_counts"))
+        .select(columns + exploded_search_columns)
+    )
 
+    # `explode` loses null values, preserve these rows
+    null_searches = (
+        dataframe
+        .where("search_counts is null")
+        .select(columns + null_search_columns)
+    )
 
-def add_profile_day(frame):
-    return frame.selectExpr(
-        "*",
-        PROFILE_COLUMN
+    # return their union
+    return exploded_searches.union(null_searches)
+
+
+def with_shares(dataframe):
+    """ Assign each client a weight for the contribution toward the
+    rollup aggregates.
+
+    Given a single profile_day with N permutations of
```

Comment:
The code no longer appears to use `profile_day` - this comment should be updated accordingly.

Also, is there test coverage for this logic?